### DISCUSSION TITLE
How to load a bgmap on Sotodlib?

### DISCUSSION THREAD

**Discussion by samdayweiss** [simonsobs/sodetlib#363] [Q&A]
I'd like to load a bg map with sotodlib so I can use it along with an axis manager generated by `load_file`. How do I do this?

**Comment by jlashner** [comment]
Hi Sam, do you think you can give a little bit of context about where you're running this? The process is different depending if you are running on a smurf-server, or an analysis computer with transferred files.

**Reply by samdayweiss** [reply]
This is on an analysis computer with transferred files (specifically on Della at Pton)

**Reply by samdayweiss** [reply]
*which is also running the entire G3tSmurf infrastructure

**Comment by jlashner** [comment]
If you have the path to the corresponding bgmap file, then you can `np.load(path)` which will give you a dict with the bgmap, polarity, etc. If you want to load for a specific set of channels, you can use the `load_bgmap` function in `sodetlib.util`, something like:
```
import sodetlib as sdl

bgmap, polarity = sdl.load_bgmap(am.det_info.band, am.det_info.channel, bgmap_file)
```

You may have to do a bit of searching to find the right bgmap file to use, it should be the last bgmap-file in the `smurf` directory before the timestream you're looking at. For site operation this will be built into the CalDb, but that probably isn't setup yet.

**Reply by jlashner** [reply]
I usually have like a custom find-function that I wrote to find files based on their file-name on simons1 (unfortunately I can't access it right now....), but you could probably do something like `find . -name *bg_map.npy` to search for all bg_map files.
